docs/articles/streams/integration.md

```diff
@@ -4,30 +4,112 @@ title: Integration
+the `CompletionStage` of the `Ask` and it will be the element that is emitted downstreams 
+from `SelectAsync`.
 
-####Source.ActorRef
+```csharp
+public class Translator : ActorBase
+{
+	protected override bool Receive(object message)
+	{
+		return message.Match()
+			.With<string>(word => 
+            {
+            	// ... process message
+				string reply = word.ToUpper();
+                // reply to the ask
+				Sender.Tell(reply, Self);
+			})
+            .WasHandled;
+	}
+}
+```
+The stream can be completed with failure by sending `Akka.Actor.Status.Failure` as reply from the actor.
+
+If the `Ask` fails due to timeout the stream will be completed with `TimeoutException` failure. 
+If that is not desired outcome you can use `Recover` on the `Ask` `CompletionStage`.
+
+If you don't care about the replies you can use `Sink.Ignore` after the `SelectAsync` stage 
+and then actor is effectively a sink of the stream.
+
+The same pattern can be used with [Actor routers](../actors/routers.md#Routers). Then you can use 
+`SelectAsyncUnordered` for better efficiency if you don't care about the order of the emitted 
+downstream elements (the replies).
+
+####Sink.ActorRefWithAck
+The sink sends the elements of the stream to the given `IActorRef` that sends back back-pressure signal.
+First element is always `OnInitMessage`, then stream is waiting for the given acknowledgement message 
+from the given actor which means that it is ready to process elements. 
+It also requires the given acknowledgement message after each stream element to make back-pressure work.
+
+If the target actor terminates the stream will be cancelled. When the stream is completed successfully 
+the  given `OnCompleteMessage` will be sent to the destination actor. When the stream 
+is completed with failure a `Akka.Actor.Status.Failure` message will be sent to the destination actor.
+
+>**Note**<br/>
+>Using `Sink.ActorRef` or ordinary `Tell` from a `Select` or `ForEach` stage means that there is 
+>no back-pressure signal from the destination actor, i.e. if the actor is not consuming the messages 
+>fast enough the mailbox of the actor will grow, unless you use a bounded mailbox with zero 
+>`mailbox-push-timeout-time` or use a rate limiting stage in front. 
+>It's often better to use `Sink.ActorRefWithAck` or `Ask` in `SelectAsync`, though. 
+
+####Source.Queue
+// NOT DONE
+`Source.Queue` can be used for emitting elements to a stream from an actor (or from anything running 
+outside the stream). The elements will be buffered until the stream can process them. You can `Offer`
+elements to  the queue and they will be emitted to the stream if there is demand from downstream, 
+otherwise they will be buffered until request for demand is received.
+
+Use overflow strategy `Akka.Streams.OverflowStrategy.Backpressure` to avoid dropping of elements 
+if the  buffer is full.
+
+`SourceQueue.Offer` returns `CompletionStage<StreamCallbackStatus<Boolean>>` 
```

Comment:
The JVM part is correct, we really return those `QueueOfferRestul`s ;-) See https://github.com/akkadotnet/akka.net/blob/v1.3/src/core/Akka.Streams/QueueOfferResult.cs 
The onyl difference is that we use `OfferAsync` + `Task`

> `ISourceQueueWithComplete.OfferAsync` returns `Task<IQueueOfferResult>`  which completes with `IQueueOfferResult.Enqueued` if element was added to buffer or sent downstream. 
It completes with `IQueueOfferResult.Dropped` if element was dropped. It can also complete with
`IQueueOfferResult.Failure` when stream failed or `IQueueOfferResult.QueueClosed` 
when downstream is completed.